src/utilities/createRouterResolve.ts

```diff
@@ -1,33 +1,28 @@
-import { RouteMethod, RouteMethodResponseImplementation, RouterRoute, Routes, isRouteMethodResponse } from '@/types'
+import { RouterRoutes } from '@/types/routerRoute'
 import { RouteWithParams, RouteWithParamsImplementation } from '@/types/routeWithParams'
 import { isRecord } from '@/utilities/guards'
-import { getRoutePath } from '@/utilities/routes'
 import { assembleUrl } from '@/utilities/urlAssembly'
 
 export type RouterResolveOptions = {
   query?: Record<string, string>,
 }
 
 export type RouterResolve<
-  TRoutes extends Routes
+  TRoutes extends RouterRoutes
 > = <
   TRoutePath extends string
->(source: string | RouteWithParams<TRoutes, TRoutePath> | ReturnType<RouteMethod>, options?: RouterResolveOptions) => string
+>(source: string | RouteWithParams<TRoutes, TRoutePath>, options?: RouterResolveOptions) => string
 
-export type RouterResolveImplementation = (source: string | RouteWithParamsImplementation | RouteMethodResponseImplementation, options?: RouterResolveOptions) => string
+export type RouterResolveImplementation = (source: string | RouteWithParamsImplementation, options?: RouterResolveOptions) => string
 
-export function createRouterResolve(routes: RouterRoute[]): RouterResolveImplementation {
+export function createRouterResolve(routes: RouterRoutes): RouterResolveImplementation {
   return (source, options) => {
     if (typeof source === 'string') {
       return source
     }
 
-    if (isRouteMethodResponse(source)) {
-      return source.url
-    }
-
     if (isRecord(source)) {
-      const match = routes.find((route) => getRoutePath(route) === source.route)
+      const match = routes.find((route) => route.name === source.route)
```

Comment:
little things like this really make this feel like the right direction.

Reply:
💯